src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

```diff
@@ -110,6 +112,16 @@ public static IResourceBuilder<AzureCosmosDBResource> RunAsEmulator(this IResour
         return builder;
     }
 
+    /// <summary>
+    /// Adds a named volume for the data folder to an Azure Cosmos DB emulator resource.
+    /// </summary>
+    /// <param name="builder">The builder for the <see cref="AzureCosmosDBEmulatorResource"/>.</param>
+    /// <param name="name">The name of the volume. Defaults to an auto-generated name based on the application and resource names.</param>
+    /// <returns>A builder for the <see cref="AzureCosmosDBEmulatorResource"/>.</returns>
+    public static IResourceBuilder<AzureCosmosDBEmulatorResource> WithDataVolume(this IResourceBuilder<AzureCosmosDBEmulatorResource> builder, string? name = null)
```

Comment:
A test can be added for this like https://github.com/dotnet/aspire/blob/68ec8eadaa49bb170a19fba99b9a17462a821118/tests/Aspire.Hosting.Redis.Tests/RedisFunctionalTests.cs#L106-L189

Reply:
I added it - however I got issues with running multiple consecutive cosmosdb emulator instances and I don't have more time to spend on this unfortunately. Would any of you be able to have a look @radical @sebastienros?